devdoc/constbuffer_array_splitter_requirements.md

```diff
@@ -0,0 +1,60 @@
+`constbuffer_array_splitter` requirements
+================
+
+## Overview
+
+`constbuffer_array_splitter` is a helper module that can take a const buffer array and transform it into another const buffer array where each const buffer is of size `max_buffer_size` (except the last one which may be smaller).
+
+For example, it is used to upload large payloads to Blob store, split across large blocks of a maximum size.
+
+## Exposed API
+
+```c
+MOCKABLE_FUNCTION(, CONSTBUFFER_ARRAY_HANDLE, constbuffer_array_splitter_split, CONSTBUFFER_ARRAY_HANDLE, buffers, uint32_t, max_buffer_size);
+```
+
+### constbuffer_array_splitter_split
+
+```c
+MOCKABLE_FUNCTION(, CONSTBUFFER_ARRAY_HANDLE, constbuffer_array_splitter_split, CONSTBUFFER_ARRAY_HANDLE, buffers, uint32_t, max_buffer_size);
+```
+
+This takes the `buffers` from a `CONSTBUFFER_ARRAY_HANDLE` and splits them into a new `CONSTBUFFER_ARRAY_HANDLE` where each const buffer is of size `max_buffer_size` (or smaller for the final remaining buffer). The caller is responsible for cleaning up the returned `CONSTBUFFER_ARRAY_HANDLE`.
+
+**SRS_OFFLOAD_STORE_BLOB_SPLITTER_42_001: [** If `buffers` is `NULL` then `constbuffer_array_splitter_split` shall fail and return `NULL`. **]**
```

Comment:
>OFFLOAD_STORE_BLOB_SPLITTER [](http://example.com/codeflow?start=6&length=27)

The specs are pointing to an old name (I think Dan fixed this in the latest master of EBS, so you might want to pull latest and copy it again). Here and everywhere

Reply:
Er, just take this branch I guess, the PR didn't go through [https://msazure.visualstudio.com/One/_git/Azure-MessagingStore/pullrequest/9807472](https://msazure.visualstudio.com/One/_git/Azure-MessagingStore/pullrequest/9807472)